src/main/java/org/spdx/library/model/license/AbstractExtractedLicenseInfo.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2022 Source Auditor Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package org.spdx.library.model.license;
+
+import org.spdx.library.InvalidSPDXAnalysisException;
+import org.spdx.library.ModelCopyManager;
+import org.spdx.library.model.SpdxInvalidTypeException;
+import org.spdx.storage.IModelStore;
+
+/**
+ * ExtractedLicenseInfo which can be represented as a concrete ExtractedLicenseInfo within
+ * the same SPDX document with all properties accessible, or as an ExternalExtractedLicenseInfo
+ * which represents a license not included within the SPDX document.
+ * 
+ * @author Gary O'Neall
+ *
+ */
+public abstract class AbstractExtractedLicenseInfo extends SimpleLicensingInfo
+implements Comparable<AbstractExtractedLicenseInfo> {
+
+	/**
+	 * Create a new ExtractedLicenseInfo using the ID
+	 * @param id
+	 * @throws InvalidSPDXAnalysisException
+	 */
+	public AbstractExtractedLicenseInfo(String id)
+			throws InvalidSPDXAnalysisException {
+		super(id);
+	}
+
+	/**
+	 * Create a new ExtractedLicenseInfo object
+	 * @param modelStore container which includes the license
+	 * @param documentUri URI for the SPDX document containing the license
+	 * @param id identifier for the license
+	 * @param copyManager if non-null, allows for copying of any properties set which use other model stores or document URI's
+	 * @param create if true, create the license if it does not exist
+	 * @throws InvalidSPDXAnalysisException 
+	 */
+	public AbstractExtractedLicenseInfo(IModelStore modelStore,
+			String documentUri, String id, ModelCopyManager copyManager,
+			boolean create) throws InvalidSPDXAnalysisException {
+		super(modelStore, documentUri, id, copyManager, create);
+	}
+
+	/* (non-Javadoc)
+	 * @see org.spdx.rdfparser.license.AnyLicenseInfo#toString()
+	 */
+	@Override
+	public String toString() {
+		// must be only the ID if we are to use this to create 
+		// parseable license strings
+		return this.getLicenseId();
+	}
+	
+	/**
+	 * @return the text
+	 * @throws SpdxInvalidTypeException 
+	 */
+	public abstract String getExtractedText() throws InvalidSPDXAnalysisException;
+
+	/**
+	 * @param text the text to set
+	 * @throws InvalidSPDXAnalysisException 
+	 */
+	public abstract void setExtractedText(String text) throws InvalidSPDXAnalysisException;
```

Comment:
Does it work to define this method only on `ExtractedLicenseInfo`? It would be nice to get rid of abstract methods that always throw for one of the extending classes

Reply:
(there are some other methods that always throw for `ExternalExtractedLicenseInfo`, but they are inherited from `SimpleLicensingInfo`, so that would require more refactoring and may be out of scope for now)

Reply:
> Does it work to define this method only on ExtractedLicenseInfo? It would be nice to get rid of abstract methods that always throw for one of the extending classes

Good catch - I removed the abstract method and implementation that throws the exception.  I originally added the abstract method in so the unit tests would pass.  But there is no reason to test that setting the extracted license text would fail.  Much better to remove the method and remove the unit test.

> there are some other methods that always throw for ExternalExtractedLicenseInfo, but they are inherited from SimpleLicensingInfo, so that would require more refactoring and may be out of scope for now

Correct - I though about addressing those, but any approach I would come up with would cause a compatibility issue - something we should take care of in the next major release of the library.